### PR TITLE
GenericMemConversionModule

### LINKED ISSUE #248: Generic Module For Memory Conversion
**Is your feature request related to a problem? Please describe.**
Need a generic Memory conversion module that can convert :-
-> Dma to Host
-> Host to Dma
-> Device to DMA
-> DMA to Device
 
**Describe the solution you'd like**
Need to pass a property that will specify output memory type

### PR DESCRIPTION
**Pull request by venkat0907**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #248 

**Description**
written a generic Memory conversion module that can convert :-
-> Dma to Host
-> Host to Dma[currently not supporting planar images]
-> Device to DMA
-> DMA to Device
-> Host to Device
-> Device to Host

**Alternative(s) considered**

Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

NO

Type Choose one: Feature

**Screenshots (if applicable)**

**Checklist**

- [x] I have read the [Contribution Guidelines](https://github.com/Apra-Labs/ApraPipes/wiki/Contribution-Guidelines)
- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

### CODE REVIEW

File: base/include/MemTypeConversion.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "Module.h"
+#include <memory>
```
Comment (yashrajsapra) [inline base/include/MemTypeConversion.h:4]: Move "<memory >" header from include to src file

File: base/include/DMAFrameUtils.h [hunk 2]
```diff
@@ -22,27 +23,89 @@ class DMAFrameUtils
         {
         case FrameMetadata::FrameType::RAW_IMAGE:
         {
+            size_t pitch[4] = {0, 0, 0, 0};
             auto rawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(metadata);
```
Comment (yashrajsapra) [inline base/include/DMAFrameUtils.h:27]: Please format all the files

**Review by yashrajsapra — CHANGES_REQUESTED** [review]

File: base/include/DMAFrameUtils.h (hunk 2 above)
Comment (venkat0907) [reply base/include/DMAFrameUtils.h:27]: sure

File: base/test/memtypeconversion_tests.cpp
[the remark this answers is not in the document]
Comment (venkat0907) [reply base/test/memtypeconversion_tests.cpp:1]: sure done

File: base/test/memtypeconversion_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+	fileReader->setNext(memconversion1);
+
+    auto memconversion2 = boost::shared_ptr<Module>(new MemTypeConversion(MemTypeConversionProps(FrameMetadata::DMABUF,stream)));
+	memconversion1->setNext(memconversion2);
+
+    auto memconversion3 = boost::shared_ptr<Module>(new MemTypeConversion(MemTypeConversionProps(FrameMetadata::HOST,stream)));
+	memconversion2->setNext(memconversion3);
+
+    auto sink = boost::shared_ptr<Module>(new FileWriterModule(FileWriterModuleProps("./data/testOutput/frame_????.raw")));
+	memconversion3->setNext(sink);
```
[the remark this answers is not in the document]
Comment (venkat0907) [reply base/test/memtypeconversion_tests.cpp:67]: added save or compare for assertion

**Review by yashrajsapra — APPROVED** [review]

**Review by mraduldubey — APPROVED** [review]

**Pull request merged**